### ISSUE TITLE
Duplicate declaration "React"

### ISSUE THREAD

**Issue by novascreen** [vslinko/babel-plugin-react-require#17]
Using this with [kesne/babel-plugin-inline-react-svg](https://github.com/kesne/babel-plugin-inline-react-svg) results in this error:
```
Module build failed: TypeError: unknown: Duplicate declaration "React"
   5 | import _inherits from 'babel-runtime/helpers/inherits';
   6 | import React from 'react';
>  7 | import React from 'react';
     |        ^
```

it seems like this plugin should add a check to prevent multiple React imports being added as mentioned here: https://github.com/kesne/babel-plugin-inline-react-svg/issues/31#issuecomment-342264348

**Comment by vslinko** [comment]
Should be closed by https://github.com/vslinko/babel-plugin-react-require/pull/18

**Comment by otakustay** [comment]
A weird problem is that `3.0.0` runs OK on my project, however the latest `3.0.1` introduces this "Duplicate declaration React" issue:

```
    ERROR in ./node_modules/@baidu/etui/node_modules/recompose/dist/Recompose.esm.js
    Module build failed (from ./node_modules/@baidu/ee-fe-tools/node_modules/babel-loader/lib/index.js):
    TypeError: /Users/otakustay/Develop/baidu/hetu/hetu-frontend/node_modules/@baidu/etui/node_modules/recompose/dist/Recompose.esm.js: Duplicate declaration "React"
    > 1 | import React, { createFactory, Component, createElement } from 'react';
        |        ^
      2 | import _extends from '@babel/runtime/helpers/esm/extends';
      3 | import shallowEqual from 'fbjs/lib/shallowEqual';
      4 | import _inheritsLoose from '@babel/runtime/helpers/esm/inheritsLoose';
```

This only appears in production mode, however `recompose` as a 3rd party lib is excluded from babel-loader, everything seems weird to me

**Comment by otakustay** [comment]
I created a repo to reproduce this issue: https://github.com/otakustay/webpack-issues/tree/react-require

Running `$(npm bin)/webpack --config=webpack.config.js` can reproduce, when `babel-plugin-react-require` is placed **before** `babel-plugin-transform-react-remove-prop-types`, this error appears, while I revert the order of these plugins, everything went OK

**Comment by otakustay** [comment]
@vslinko This is not solved in #18 , do we have any chance to dig this issue again?

**Comment by jeremejevs** [comment]
The problem comes from `babel-plugin-transform-react-remove-prop-types`'s usage of `programPath.scope.crawl`, which isn't compatible with the hack used in version 3 of this plugin. It makes Babel "re-scan" the file, find two React imports, and blow up, before `react-require` gets a chance to clean up. A potential fix would be to try moving everything into `Program.enter`, but I'm not willing to invest time into that, I'm afraid. In the meanwhile, it's a valid strategy to just keep the plugins listed in the order that works (this isn't uncommon with Babel) or revert to version 2.

**Comment by behzadht** [comment]
Any news on this issue? Still getting the error described in the initial message. The issue is not solved by #18 as suggested by @vslinko.

**Comment by MuhammadZaidSE** [comment]
Facing the same issue
```
| import React from "react";
> import React from "react";
| 
```

Using
`"babel-plugin-inline-react-svg": "^0.4.0"`
`"next": "^7.0.2"`

babelrc config
```
{
  "presets": [
    "next/babel"
  ],
  "plugins": [
    ["styled-components", { "ssr": true, "displayName": true, "preprocess": false } ],
    ["module-resolver", {
      "root": ["./"],
      "alias": {
        "lib": "./lib",
        "pages": "./pages",
        "components": "./components",
        "containers": "./containers",
        "store": "./store",
        "layouts": "./layouts",
        "static": "./static",
        "themes": "./themes",
        "utils": "./utils"
      }
    }],
    [
      "inline-react-svg",
      {
        "svgo": {
          "plugins": [
            {
              "removeAttrs": { "attrs": "(data-name)" }
            },
            {
              "cleanupIds": true
            }
          ]

        }
      }
    ],
    [
      "inline-import",
      {
        "extensions": [".css"]
      }
    ]
  ],
  "env": {
    "development": {
      "plugins": ["dotenv-import"]
    },
    "production": {
      "plugins": [
        ["dotenv-import", {
          "path": ".env.production"
        }]
      ]
    }
  }
}
```

kindly help me

**Comment by vslinko** [comment]
I have published version `3.0.2` that should fix @otakustay case.
@otakustay please answer how it works for you.

To everyone else: Please try this patch and create an example repo If you still have an error.

**Comment by adamsoffer** [comment]
Was this resolved? I'm still getting this error with `babel-plugin-inline-react-svg`

**Comment by b2times4** [comment]
+1 Still getting this error

**Comment by adamsoffer** [comment]
@vslinko - this is still an issue. See https://github.com/airbnb/babel-plugin-inline-react-svg/issues/50#issuecomment-569441842

Mind reopening?

**Comment by vslinko** [comment]
@adamsoffer should be fixed in 3.1.3
